packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -356,13 +346,57 @@ export class FieldSchema<
 		 */
 		public readonly props?: FieldProps<TCustomMetadata>,
 	) {
+		if (!(this instanceof FieldSchemaAlpha)) {
+			throw new UsageError("FieldSchema is @sealed: sub-classing is not allowed.");
+		}
+
 		this.lazyTypes = new Lazy(() => normalizeAllowedTypes(this.allowedTypes));
 		// TODO: optional fields should (by default) get a default provider that returns undefined, removing the need to special case them here:
 		this.requiresValue =
 			this.props?.defaultProvider === undefined && this.kind !== FieldKind.Optional;
 	}
 }
 
+/**
+ * {@link FieldSchema} including alpha APIs (currently {@link SimpleFieldSchema}).
+ * @remarks
+ * This class will go away once the alpha APIs are stable and implemented by {@link FieldSchema}.
+ * @sealed @alpha
+ */
+export class FieldSchemaAlpha<
```

Comment:
Maybe I'm just blind, but I can't tell by scanning the diff - where do we construct these?

Reply:
All FieldSchema are constructed via the non exported createFieldSchema function. Public users use SchemaFactory which calls that. This pattern makes it possible to export this class, but not the factory function for it.